src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java

```diff
@@ -123,6 +127,9 @@ protected DeleteModelNodeResponse nodeOperation(DeleteModelNodeRequest request)
 
         cache.get().clear(adID);
 
+        // delete realtime task cache
+        adTaskCacheManager.removeRealtimeTaskCache(adID);
```

Comment:
TODO: cleanup stale realtime task cache in hourly cron. 

Will add this in another PR.